src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.TrailingHeaders.cs

```diff
@@ -217,5 +225,124 @@ public async Task GetAsync_NoTrailingHeaders_EmptyCollection()
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.SupportsAlpn))]
+        public async Task Http2GetAsync_TrailerHeaders_TrailingPseudoHeadersThrow()
+        {
+            using (var server = Http2LoopbackServer.CreateServer())
+            using (var client = new HttpClient(CreateHttpClientHandler(useSocketsHttpHandler: true, useHttp2LoopbackServer: true)))
+            {
+                Task<HttpResponseMessage> sendTask = client.GetAsync(server.Address);
+
+                await server.EstablishConnectionAsync();
+
+                int streamId = await server.ReadRequestHeaderAsync();
+
+                // Response header.
+                await server.SendDefaultResponseHeadersAsync(streamId);
+                await server.WriteFrameAsync(MakeDataFrame(streamId, DataBytes));
+                // Additional trailing header frame with pseudo-headers again..
+                await server.SendResponseHeadersAsync(streamId, hasStatusCode:true, headers: TrailingHeaders, endStream : true);
+
+                await Assert.ThrowsAsync<HttpRequestException>(() => sendTask);
+            }
+        }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.SupportsAlpn))]
+        public async Task Http2GetAsyncResponseHeadersReadOption_TrailingHeaders_Available()
+        {
+            using (var server = Http2LoopbackServer.CreateServer())
+            using (var client = new HttpClient(CreateHttpClientHandler(useSocketsHttpHandler: true, useHttp2LoopbackServer: true)))
+            {
+                Task<HttpResponseMessage> sendTask = client.GetAsync(server.Address, HttpCompletionOption.ResponseHeadersRead);
+
+                await server.EstablishConnectionAsync();
+
+                int streamId = await server.ReadRequestHeaderAsync();
+
+                // Response header.
+                await server.SendDefaultResponseHeadersAsync(streamId);
+
+                // Response data, missing Trailers.
+                await server.WriteFrameAsync(MakeDataFrame(streamId, DataBytes));
+
+                HttpResponseMessage response = await sendTask;
+                Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+                // Pending read on the response content.
+                Console.WriteLine(response.TrailingHeaders);
+                Assert.Empty(response.TrailingHeaders);
+
+                Stream stream = await response.Content.ReadAsStreamAsync();
+                Byte[] data = new Byte[100];
+                await stream.ReadAsync(data, 0, data.Length);
+
+                // Intermediate test - haven't reached stream EOF yet.
+                Assert.Empty(response.TrailingHeaders);
+
+                // Finish data stream and write out trailing headers.
+                await server.WriteFrameAsync(MakeDataFrame(streamId, DataBytes));
+                await server.SendResponseHeadersAsync(streamId, hasStatusCode:false, headers: TrailingHeaders, endStream : true);
```

Comment:
nit: The spacing here is inconsistent. Also even though these are named parameters, it would be nice if they were in the same order as the declared parameters.